### PR TITLE
cypress test cases added

### PR DESCRIPTION
**Pull request by shahrukh-ijaz**
cypress testing module is added

### CODE REVIEW

File: cypress/integration/Test_Phonebook/test_login.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+describe('Login Test', function(){
```
Comment (asamolion) [inline cypress/integration/Test_Phonebook/test_login.js]: Add assertions to this test case to validate the business logic of the application.

File: cypress/integration/Test_Phonebook/test_register.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+describe('Login Test', function(){
```
Comment (asamolion) [inline cypress/integration/Test_Phonebook/test_register.js]: Use the proper name for this test suite.

File: cypress/integration/Test_Phonebook/test_stub_login.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// cypress/integration/test_stub_login.js
+
+describe("Testing the widget", function() {
+    it("Widget does something with AJAX / XHR", function() {
+        cy
+            .request('POST', 'http://127.0.0.1:8001/api/login', { username: 'shahrukh-ijaz', password: 'shahrukh31' })
+            .then((response) => {
+                // response.body is automatically serialized into JSON
+               expect(response.body).to.have.property('token', '8a404e47938e966ff6d4b219db236c1e5c02a4c7') // true
```
Comment (asamolion) [inline cypress/integration/Test_Phonebook/test_stub_login.js]: Are you sure that this token will be the one generated each time?

File: cypress/integration/Test_Phonebook/test_login.js (hunk 1 above)
Comment (shahrukh-ijaz) [reply cypress/integration/Test_Phonebook/test_login.js]: done

File: cypress/integration/Test_Phonebook/test_register.js (hunk 2 above)
Comment (shahrukh-ijaz) [reply cypress/integration/Test_Phonebook/test_register.js]: done

File: cypress/integration/Test_Phonebook/test_stub_login.js (hunk 3 above)
Comment (shahrukh-ijaz) [reply cypress/integration/Test_Phonebook/test_stub_login.js]: done

File: cypress/integration/Test_Phonebook/test_login.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            cy.get('.navbar-header').click();
+            cy.get('a#profile_link').click();
+            cy.get('a#home_link').click().then(response=>{
+                cy.get('a#addContact_link').click();
+                cy.get('input#first_name').type('osama');
+                cy.get('input#last_name').type('arshad');
+                cy.get('input#note').type('team lead contact number');
+                cy.get('input#dob').click().type('2019-01-01');
+                cy.get('button[type="submit"]').click().then(response=>{
+                    cy.get('a#home_link').click().then(response=>{
```
Comment (asamolion) [inline cypress/integration/Test_Phonebook/test_login.js]: Do not cascade your get calls. Directly call the element required.

File: cypress/integration/Test_Phonebook/test_register.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        // expect(true).to.equal(true);
+        cy.visit('http://localhost:3006/register')
+
+        cy.get('input#email').type('ali_1@gmail.com');
+        cy.get('input#password').type('ali123');
+        cy.get('input#username').type('ali123');
+        cy.get('input#first_name').type('ali');
+        cy.get('input#last_name').type('raza');
+        cy.get('button[type="submit"]').click().then(response=>{
+            expect(true).to.be.true
```
Comment (asamolion) [inline cypress/integration/Test_Phonebook/test_register.js]: Seriously? Add some real assertions please. Test whether the response is good. the status code. the token. whether the token is being stored in the localStorage etc.

**Pull request merged**